payjoin/src/v2.rs

```diff
@@ -193,10 +193,17 @@ pub fn ohttp_encapsulate(
 ) -> Result<(Vec<u8>, ohttp::ClientResponse), Error> {
     let ctx = ohttp::ClientRequest::from_config(ohttp_keys)?;
     let url = url::Url::parse(target_resource)?;
+    let authority_bytes = url.host().map_or_else(Vec::new, |host| {
+        let mut authority = host.to_string();
+        if let Some(port) = url.port() {
+            authority.push_str(&format!(":{}", port));
+        }
+        authority.into_bytes()
+    });
```

Comment:
This is super annoying but the `url.authority()` function was removed in url v2.3. It was brought back in v2.5 and we should bring it back then, too